packages/nimble-components/src/table/tests/table-action-menu.spec.ts

```diff
@@ -516,6 +615,33 @@ describe('Table action menu', () => {
             expect(currentSelection).toEqual([
                 simpleTableData[rowIndex].stringData
             ]);
+            expect(getEmittedRecordIdsFromSpy(toggleListener.spy)).toEqual(
+                jasmine.arrayWithExactContents(currentSelection)
+            );
+        });
+
+        it('clicking action menu button for an already selected row keeps entire selection selected', async () => {
+            await element.setSelectedRecordIds([
+                simpleTableData[0].stringData,
+                simpleTableData[2].stringData
+            ]);
+
+            const toggleListener = createEventListener(
+                element,
+                'action-menu-toggle'
+            );
+            pageObject.setRowHoverState(2, true);
+            await pageObject.clickCellActionMenu(2, 0);
+            await toggleListener.promise;
+
+            const currentSelection = await element.getSelectedRecordIds();
```

Comment:
Do we have tests that verify the event data for the action menu toggle events contains the correct list of selected records too?

Reply:
Yes, if I understand what you're asking correctly, this test is actually one of the tests verifying that the event data has the correct set of records.

But, I could/should improve these tests by asserting that both the toggle & the beforetoggle events contain the correct list of selected records.

Reply:
You understood correctly: I'm asking for those assertions on toggle and beforetoggle event data too.